rasterio/plot.py

```diff
@@ -50,6 +50,9 @@ def show(source, cmap='gray', with_bounds=True):
         rather than pixel coordinates. Only works when source is
         (raster dataset, bidx).
     """
+    if plt is None:  # pragma: no cover
+        raise ImportError("Could not import matplotlib")
+
```

Comment:
I think using a "lazy import" so that matplotlib remains an optional dependency is an accepted design pattern. Moving this into a separate get_plt() function so it can be used from multiple plotting functions only makes sense. I guess I don't understand #741 enough to figure where it fits into that.